service.py

```
@@ -1,116 +1,420 @@
+                    timeout=30
+                )
+                response.raise_for_status()
+                
+                self.trips = response.json()
+                logger.info(f"Successfully fetched {len(self.trips)} trips")
+                return
+                
+            except requests.exceptions.RequestException as e:
+                logger.error(f"Get trips attempt {attempt + 1} failed: {e}")
+                if attempt == 2:  # Last attempt
+                    raise APIException(f"Failed to get trips after 3 attempts: {e}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)